File: api/openai.py
import openai
import os

# Fetch the API key from the environment variables
openai.api_key = os.getenv('OPENAI_API_KEY')

if not openai.api_key:
    raise ValueError("API key is not set. Please set the OPENAI_API_KEY environment variable.")

response = openai.ChatCompletion.create(
    model="gpt-4",
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "What's the weather like today?"}
    ]
)

print(response['choices'][0]['message']['content'])

File: api/openai.js
const openai = require('openai');
const dotenv = require('dotenv');
dotenv.config();

openai.apiKey = process.env.OPENAI_API_KEY;

const getOpenAIResponse = async (prompt) => {
    try {
        const response = await openai.Completion.create({
            model: "text-davinci-002",
            prompt: prompt,
            max_tokens: 150
        });
        return response.choices[0].text.trim();
    } catch (error) {
        console.error("Error calling OpenAI API:", error);
        return null; 
    }
}

module.exports = { getOpenAIResponse };